### PR TITLE
Add CreateRequest methods

### PR DESCRIPTION
**Pull request by m-nash**
Fixes https://github.com/microsoft/typespec/issues/3679
Fixes https://github.com/microsoft/typespec/issues/3826

### CODE REVIEW

File: packages/http-client-csharp/generator/Microsoft.Generator.CSharp.ClientModel/src/Providers/RestClientProvider.cs [hunk 1]
```diff
@@ -1,43 +1,276 @@
...
+
         protected override MethodProvider[] BuildMethods()
         {
             List<MethodProvider> methods = new List<MethodProvider>();
 
             foreach (var operation in _inputClient.Operations)
             {
-                var methodProvider = (ScmMethodProviderCollection)ClientModelPlugin.Instance.TypeFactory.CreateMethods(operation, ClientProvider);
-                methods.Add(methodProvider.BuildCreateMessageMethod());
+                //var methodProvider = (ScmMethodProviderCollection)ClientModelPlugin.Instance.TypeFactory.CreateMethods(operation, ClientProvider);
```
Comment (jorgerangel-msft) [inline packages/http-client-csharp/generator/Microsoft.Generator.CSharp.ClientModel/src/Providers/RestClientProvider.cs:101]: nit: did we mean to remove this?

**Review by jorgerangel-msft — APPROVED** [review]

File: packages/http-client-csharp/generator/Microsoft.Generator.CSharp.ClientModel/src/Providers/RestClientProvider.cs [hunk 2]
```diff
@@ -1,43 +1,276 @@
...
+        }
+
+        protected override FieldProvider[] BuildFields()
+        {
+            return
+            [
+                _pipelineMessageClassifier200,
+                _pipelineMessageClassifier204,
+                _pipelineMessageClassifier2xxAnd4xx
+            ];
```
Comment (ShivangiReja) [inline packages/http-client-csharp/generator/Microsoft.Generator.CSharp.ClientModel/src/Providers/RestClientProvider.cs:87]: Can we add some test cases for this?

File: packages/http-client-csharp/generator/Microsoft.Generator.CSharp.ClientModel/src/Providers/ClientProvider.cs [hunk 3]
```diff
@@ -40,29 +46,33 @@ protected override PropertyProvider[] BuildProperties()
...
         {
             return
             [
                 new ConstructorProvider(
                     new ConstructorSignature(Type, $"{_inputClient.Description}", MethodSignatureModifiers.Public, []),
-                    new MethodBodyStatement[] { PipelineProperty.Assign(ClientPipelineSnippets.Create()).Terminate() },
+                    new MethodBodyStatements(
+                    [
+                        PipelineProperty.Assign(ClientPipelineSnippets.Create()).Terminate(),
+                        EndpointField.Assign(New.Instance(typeof(Uri), Literal("http://foo.com"))).Terminate()
```
Comment (ArcturusZhang) [inline packages/http-client-csharp/generator/Microsoft.Generator.CSharp.ClientModel/src/Providers/ClientProvider.cs:62]: I think this is only a placeholder?

File: packages/http-client-csharp/generator/Microsoft.Generator.CSharp.ClientModel/src/Providers/RestClientProvider.cs [hunk 4]
```diff
@@ -1,43 +1,275 @@
...
+        {
+            List<ParameterProvider> methodParameters = new();
+            foreach (InputParameter inputParam in operation.Parameters)
+            {
+                if (inputParam.Kind != InputOperationParameterKind.Method)
+                    continue;
+
+                ParameterProvider parameter = inputParam.Location == RequestLocation.Body
+                    ? ScmKnownParameters.BinaryContent
+                    : ClientModelPlugin.Instance.TypeFactory.CreateParameter(inputParam);
```
Comment (ArcturusZhang) [inline packages/http-client-csharp/generator/Microsoft.Generator.CSharp.ClientModel/src/Providers/RestClientProvider.cs:269]: I am curious how this method works.
Here we excluded the body parameter - but for path parameters, in protocol methods and in convenience methods they might have different types for the same input parameters, for instance when its type is an enum?

File: packages/http-client-csharp/generator/Microsoft.Generator.CSharp/src/Writers/CodeWriter.cs [hunk 5]
```diff
@@ -559,6 +559,8 @@ private void AppendType(CSharpType type, bool isDeclaration, bool writeTypeNameO
                 AppendRaw("global::");
                 AppendRaw(type.Namespace);
                 AppendRaw(".");
+                if (type.DeclaringType is not null)
+                    AppendRaw($"{type.DeclaringType.Name}.");
                 AppendRaw(type.Name);
```
Comment (ArcturusZhang) [inline packages/http-client-csharp/generator/Microsoft.Generator.CSharp/src/Writers/CodeWriter.cs:564]: would it be more reasonable if write this in this way:
```csharp
if (type.DeclaringType is {} declaringType)
{
	writeTypeName(declaringType);
}
else
{
	AppendRaw("global::");
        AppendRaw(type.Namespace);
        AppendRaw(".");
}
AppendRaw(".");
AppendRaw(type.Name);
```

File: packages/http-client-csharp/generator/Microsoft.Generator.CSharp.ClientModel/src/Providers/ClientProvider.cs (hunk 3 above)
Comment (m-nash) [reply packages/http-client-csharp/generator/Microsoft.Generator.CSharp.ClientModel/src/Providers/ClientProvider.cs:62]: yes its already replaced by Jorge today https://github.com/microsoft/typespec/blob/main/packages/http-client-csharp/generator/TestProjects/Local/Unbranded-TypeSpec/src/Generated/UnbrandedTypeSpecClient.cs#L46

File: packages/http-client-csharp/generator/Microsoft.Generator.CSharp.ClientModel/src/Providers/RestClientProvider.cs (hunk 4 above)
Comment (m-nash) [reply packages/http-client-csharp/generator/Microsoft.Generator.CSharp.ClientModel/src/Providers/RestClientProvider.cs:269]: good call out opened a follow up issue here https://github.com/microsoft/typespec/issues/3953

File: packages/http-client-csharp/generator/Microsoft.Generator.CSharp/src/Writers/CodeWriter.cs (hunk 5 above)
Comment (m-nash) [reply packages/http-client-csharp/generator/Microsoft.Generator.CSharp/src/Writers/CodeWriter.cs:564]: what does `writeTypeName` do here?

**Pull request merged**